src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -649,24 +589,16 @@ private static bool ValidateUrlEncodingParameters(byte[] bytes, int offset, int
 
         private static bool StringRequiresHtmlDecoding(string s)
         {
-            if (s_htmlDecodeConformance == UnicodeDecodingConformance.Compat)
+            // this string requires html decoding if it contains '&' or a surrogate character
+            for (int i = 0; i < s.Length; i++)
             {
-                // this string requires html decoding only if it contains '&'
-                return (s.IndexOf('&') >= 0);
-            }
-            else
-            {
-                // this string requires html decoding if it contains '&' or a surrogate character
-                for (int i = 0; i < s.Length; i++)
+                char c = s[i];
+                if (c == '&' || Char.IsSurrogate(c))
```

Comment:
It does, but I stayed away from doing any sweeping changes around formatting. The only change to this code was indentation, as I removed the whole if block and unindented the else.